TYPO3.Flow/Classes/TYPO3/Flow/Http/Component/TrustedProxiesComponent.php

```diff
@@ -50,14 +50,21 @@ public function handle(ComponentContext $componentContext)
         }
 
         $hostHeader = $this->getFirstTrustedProxyHeaderValue(self::HEADER_HOST, $trustedRequest);
+        $portFromHost = null;
         if ($hostHeader !== null) {
+            $portSeparator = strrpos($hostHeader, ':');
+            if ($portSeparator > 0) {
```

Comment:
Hm, yeah. I had that exact solution with explode first, but thought it would lead to wrong results if there are two `:` in the host name. Not sure if that would ever be the case, as that would at most be valid when specifying `user:password@host` and that's nothing that should ever occur in a host header, no? Anyway, I'm open to change this in good faith :)